omx/gstomxcamera.c

```diff
@@ -1071,3 +1125,49 @@ gst_omx_camera_create (GstPushSrc * src, GstBuffer ** buf)
     return ret;
   }
 }
+
+static GstClock *
+gst_omx_camera_provide_clock (GstElement * element)
+{
+  GstOMXCamera *self = GST_OMX_CAMERA (element);
+  GstClock *clock;
+
+  g_return_val_if_fail (GST_IS_OMX_CLOCK (self->clock), NULL);
+
+  GST_OBJECT_LOCK (self);
+  if (!self->provide_clock)
+    goto clock_disabled;
+
+  clock = GST_CLOCK_CAST (gst_object_ref (self->clock));
+
+  GST_OBJECT_UNLOCK (self);
+
+
+  return clock;
```

Comment:
In the case of `clock_disabled` you are returning an uninitialized `GstClock *`. You should initialize `clock` to NULL. This should've given you a warning.

Reply:
You are right, already fixed!